### PR TITLE
[WIP] Verify candidate block header

### LINKED ISSUE #1243: consensus:  Verify the candidate block header before submitting a validation vote.
**Describe the bug**
Prior to casting a validation vote, a committee member is currently executing `VM:VerifyStateTransition` only. However, to be consistent with the block acceptance procedure, a provisioner must also ensure that the candidate block comes with a fully valid block header. 

**To Reproduce**
Propagate a candidate with valid transactions set but invalid block header.

### PR DESCRIPTION
**Pull request by goshawk-3**
fixes #1243

### CODE REVIEW

File: consensus/src/validation/step.rs [hunk 1]
```diff
@@ -54,20 +54,41 @@ impl<T: Operations + 'static> ValidationStep<T> {
...
+            .verify_block_header(candidate.header(), true)
+            .await
+        {
+            error!(event = "invalid_header", ?err);
+            return;
+        };
+
+        // Call Verify State Transition to make sure transactions set is valid
+        if let Err(err) = Self::call_vst(candidate, ru, executor).await {
+            error!(event = "failed_vst_call", ?err);
```
Comment (herr-seppia) [inline consensus/src/validation/step.rs:79]: Should this `return`?

Comment (herr-seppia) [reply consensus/src/validation/step.rs:79]: Then you should add the "return" otherwise cast_vote is called

Comment (goshawk-3) [reply consensus/src/validation/step.rs:79]: Good catch!

File: consensus/src/validation/step.rs [hunk 2]
```diff
@@ -54,20 +54,41 @@ impl<T: Operations + 'static> ValidationStep<T> {
...
-                );
-                return;
-            }
+        // A Validation step with empty/default Block produces a Nil Vote
+        if hash == [0u8; 32] {
+            Self::cast_vote([0u8; 32], ru, iteration, outbound, inbound).await;
+            return;
+        }
+
+        // Verify candidate header all fields except the winning certificate
```
Comment (fed-franz) [inline consensus/src/validation/step.rs:65]:
```suggestion
        // Verify candidate header (all fields except the winning certificate)
```

**Review by fed-franz — APPROVED** [review]

File: consensus/src/validation/step.rs [hunk 3]
```diff
@@ -54,32 +54,33 @@ impl<T: Operations + 'static> ValidationStep<T> {
         inbound: AsyncQueue<Message>,
         executor: Arc<Mutex<T>>,
     ) {
-        let hash = candidate.header().hash;
+        let header = candidate.header();
 
         // A Validation step with empty/default Block produces a Nil Vote
-        if hash == [0u8; 32] {
+        if header.hash == [0u8; 32] {
```
Comment (fed-franz) [inline consensus/src/validation/step.rs:60]: this is not necessary. you can just check header+vst if hash!=0 and then call `cast_vote` with `header.hash`
ie:
if header.hash != [0u8; 32]: 
  check header
  check vst
  
cast_vote(header.hash)

Just saying, feel free to keep it this way if you prefer

**Review by fed-franz — APPROVED** [review]

File: consensus/src/validation/step.rs (hunk 3 above)
Comment (goshawk-3) [reply consensus/src/validation/step.rs:60]: Current impl highlights the logic of voting NiL if the provided block is empty

**Pull request merged**